src/core/manager.rs

```diff
@@ -734,6 +765,51 @@ impl<'a> WindowManager<'a> {
         }
     }
 
+    /// Get a vector of references to the Clients found matching 'selector'
+    pub fn all_clients(&self, selector: &Selector<Client>) -> Vec<&Client> {
+        match selector {
+            Selector::Focused => self.focused_client().into_iter().collect(),
+            Selector::WinId(id) => self.client_map.get(&id).into_iter().collect(),
+            Selector::Condition(f) => self
+                .client_map
+                .iter()
+                .filter(|(_, v)| f(v))
```

Comment:
This filter and map should be doable with a `flat_map` returning Options (see ` keycodes_from_xmodmap` in `helpers` for an example) I think :slightly_smiling_face: